spot_controllers/include/spot_controllers/forward_state_controller.hpp

```diff
@@ -0,0 +1,57 @@
+// File modified. Modifications Copyright (c) 2024 Boston Dynamics AI Institute LLC.
+// All rights reserved.
+
+// --------------------------------------------------------------
+// Copyright 2020 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "forward_command_controller/forward_command_controller/forward_controllers_base.hpp"
+#include "forward_state_controller_parameters.hpp"  // NOLINT(build/include_subdir)
+#include "spot_controllers/visibility_control.h"
+
+namespace spot_controllers {
+/**
+ * \brief Forward command controller for a set of interfaces.
+ *
+ * This class forwards the command signal for a set of interfaces over a set of joints.
+ *
+ * \param joints Names of the joint to control.
+ * \param interface_names Names of the interfaces to command.
+ *
+ * Subscribes to:
+ * - \b commands (std_msgs::msg::Float64MultiArray) : The commands to apply.
+ */
+class ForwardStateController : public forward_command_controller::ForwardControllersBase {
```

Comment:
open to suggestions of potential better names than ForwardStateController. The main idea is just that it takes in a list of joints and a list of interfaces. 
Builtin controllers already have ForwardCommandController (list of joints, one interface) and MultiInterfaceForwardCommandController (one joint, list of interfaces) https://github.com/ros-controls/ros2_controllers/blob/humble/forward_command_controller/forward_command_plugin.xml

Reply:
I think ForwardStateController is fine